### PR TITLE
Rework repo buttons (#20602, #20718)

### PR DESCRIPTION
**Pull request by silverwind**
Backport #20602 and #20718 to 1.17. Translations can be backported separately.

### CODE REVIEW

**Review by Gusted — APPROVED** [review]

File: templates/repo/home.tmpl [hunk 1]
```diff
@@ -60,55 +60,52 @@
...
-						{{end}}
 					{{end}}
-					{{if and (ne $n 0) (not .IsViewFile) (not .IsBlame) }}
-						<a href="{{.RepoLink}}/commits/{{.BranchNameSubURL}}/{{.TreePath | PathEscapeSegments}}" class="ui button">
-							{{.i18n.Tr "repo.file_history"}}
-						</a>
-					{{end}}
-				</div>
-
+					<a href="{{.Repository.Link}}/find/{{.BranchNameSubURL}}" class="ui compact basic button tooltip" data-content="{{.i18n.Tr "repo.find_file.go_to_file"}}">{{svg "octicon-file-moved" 15}}</a>
```
Comment (Gusted) [inline templates/repo/home.tmpl:75]: This backport doesn't include the changes from https://github.com/go-gitea/gitea/pull/20718?

Comment (silverwind) [reply templates/repo/home.tmpl:75]: No, but I will add them here now that that's merged.

Comment (silverwind) [reply templates/repo/home.tmpl:75]: included

**Review by jolheiser — APPROVED** [review]

**Comment by zeripath** [comment]
>  Translations can be backported separately.

I think that is a recipe for forgetting to backport them at all. We should do this now assuming that a good number of translations have `editor.add_file` set in them in main

**Comment by zeripath** [comment]
Pushed to 1.17.2 - not enough translations ready.

**Comment by silverwind** [comment]
Yeah, will check for translations

**Comment by silverwind** [comment]
@jolheiser was so nice to backport the translations. We can land this if we land https://github.com/go-gitea/gitea/pull/20834 immediately afterwards.

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**